# Corpus Language Test
The sole purpose of this notebook is to verify the following hypothesis about the clickbait corpus:
> All post titles are English.

In [1]:
import pandas as pd
from os import path
import langdetect

pd.set_option('display.max_colwidth', 1000)

## Load data

In [2]:
data_path = "../data"
dataset = "clickbait17-train-170331"

In [3]:
df = pd.read_json(path.join(data_path, dataset, "instances.jsonl"), lines=True, encoding='utf8')

## Language check
We use the `langdetect` Python package to determine the language of the post title.

In [4]:
df['lang_postText'] = df['postText'].apply(lambda x: langdetect.detect(x[0]))
non_english = df[df['lang_postText'] != 'en']

Let's first check how many entries are non-English.

In [5]:
print("{} / {}".format(len(non_english), len(df)))

33 / 2459


Just ~30 out of 2459 entries are non-English. Now we manually verify if the detection was correct.

In [6]:
non_english[['postText', 'lang_postText']]

postText  \
35                                                                        [How do dogs donate blood?]   
239                                            [Mexico elects first independent governor 'El Bronco']   
293                                                     [I’ve hit a sleep wall and I’m seeing double]   
300                                                                             [Are you a “yuccie?”]   
321                                  [India frustrated against Bangladesh as Ajinkya Rahane makes 98]   
466                                                       [MORE: WWE legend #DustyRhodes dies at 69.]   
550   [Look inside Apple's swanky new Upper East Side store in NYC Old bank vault is a VIP showroom:]   
653                                                   [37 difficult questions from my mixed-race son]   
686                                                        [Harriet Harman v David Cameron at #PMQs:]   
764                                                             [Teenager denies murdering boy of 15]   
768                                       [Five hurt after smoke in cabin triggers emergency landing]   
822                             [FIFA's lame Sepp Blatter movie made $607 in its U.S. debut weekend:]   
890                                                       [Could your imagination be making you FAT?]   
950                                                               [Delays likely as rail work begins]   
962                                                     [10 unique places to hang your flatscreen TV]   
1053                             [Saudi-led airstrikes in Yemen kill at least 5, damage UNESCO site.]   
1094                               [Belarus v Spain: Euro 2016 qualifier – live! via @guardian_sport]   
1175                      [A question of trust? Anderson report lays out tests for surveillance laws]   
1278                                         [Charles Manson prosecutor Vincent Bugliosi dies at 80.]   
1296                                     [Magna Carta celebrations reignite a debate over its legacy]   
1434                                                              [Honour for Alan Barnes fundraiser]   
1462                                         [Obama urges Cameron to meet 2% defence spending target]   
1480                                                             [Happy Birthday, Sasha Obama! ? D Q]   
1513                           [Wales v Belgium: European 2016 qualifier – live! via @guardian_sport]   
1649                                                 [Escaped circus elephant kills a man in Germany]   
1661                                      [Charles Manson prosecutor Vincent Bugliosi dies at age 80]   
1680                                                              [Magna Carta celebrations to begin]   
1852                                                                       [You do you, @MileyCyrus.]   
1964                                                                             [Are you yuccie AF?]   
2035                                                                             [Stocks are sliding]   
2053                                                [Water park under fire for Muslim-friendly event]   
2306                             [WATCH LIVE: President Barack Obama speaks at G-7 summit in Germany]   
2415                                                                                    [Like a boss]   

     lang_postText  
35              af  
239             es  
293             af  
300             es  
321             id  
466             af  
550             af  
653             fr  
686             sv  
764             af  
768             da  
822             af  
890             tl  
950             af  
962             fr  
1053            da  
1094            it  
1175            fr  
1278            fr  
1296            ca  
1434            fr  
1462            nl  
1480            so  
1513            ca

We observe that all post titles are misclassfied as non-English, and are therefore certain enough that the entire corpus is indeed English.